File: 백준/Bronze/24578. Ultimate Binary Watch/Ultimate Binary Watch.cc
#include <stdio.h>

int main(void) {
    char digits[5];
    scanf("%s", digits);

    for (int i = 3; i >= 0; i--) {
        for (int j = 0; j < 4; j++) {
            if (j == 1 || j == 3) printf(" ");
            else if (j == 2) printf("   ");
            if (((digits[j] - '0') & (1 << i)) >> i) printf("*");
            else printf(".");
        }
        printf("\n");
    }
    return 0;
}

File: 백준/Bronze/24578. Ultimate Binary Watch/Ultimate Binary Watch.py
import sys
input = sys.stdin.readline

digits = input().rstrip()

for i in range(4):
    for j in range(4):
        print('*' if ((int(digits[j])) & (1 << (3-i))) >> (3-i) else '.', end='')
        if j == 1: print('   ', end='')
        elif j != 3: print(' ', end='')
    print()
